client/src/app/shared/components/Notifications.tsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import {
+  Alert,
+  AlertActionCloseButton,
+  AlertGroup,
+} from "@patternfly/react-core";
+import { NotificationsContext } from "../notifications-context";
+
+export type notificationType =
+  | "success"
+  | "info"
+  | "warning"
+  | "danger"
+  | "default";
+
+export interface INotification {
+  title: string;
+  variant: notificationType;
+  key?: string;
+  message?: string | JSX.Element;
+  actionClose?: boolean;
```

Comment:
It seems like (almost?) every use of `pushNotification()` has `actionClose: true`, I wonder if that should be a default? we could swap it to a `hideCloseButton` boolean that is usually excluded, maybe.

It also seems strange to call this `actionClose` as a boolean since the `actionClose` prop of Alert itself is a ReactNode.

Reply:
Oh, also, for `message` you can use `React.ReactNode` as the type instead of `string | JSX.Element`. ReactNode will allow anything that can be a JSX child.

Reply:
@mturley, yeah I was wondering about a default `actionClose` too. It's now wrapped through a default notification.

I remember you and I having that discussion about JSX.Element vs ReactNode in my early days with React ;) 
A `React.ReactNode` is better for `message` and `actionClose`.